handheld/src/main/java/org/jraf/android/bikey/app/mediabutton/MediaButtonHandler.java

```diff
@@ -0,0 +1,161 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.jraf.android.bikey.app.mediabutton;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.preference.PreferenceManager;
+import android.support.annotation.MainThread;
+import android.support.v4.media.session.MediaSessionCompat;
+import android.view.KeyEvent;
+
+import org.jraf.android.bikey.R;
+import org.jraf.android.bikey.app.collect.LogCollectorService;
+import org.jraf.android.bikey.backend.provider.ride.RideState;
+import org.jraf.android.bikey.backend.ride.RideManager;
+import org.jraf.android.bikey.common.Constants;
+import org.jraf.android.util.log.Log;
+import org.jraf.android.util.string.StringUtil;
+
+public class MediaButtonHandler {
+
+    private static final String TAG = MediaButtonHandler.class.getSimpleName();
+    private MediaSessionCompat mMediaSessionCompat;
+
+    private static final MediaButtonHandler INSTANCE = new MediaButtonHandler();
+
+    private MediaButtonHandler() {
+        // prevent instantiation
+    }
+
+    public static MediaButtonHandler get() {
+        return INSTANCE;
+    }
+
+    @MainThread
+    public void start(Context context) {
+        Log.d();
+        if (mMediaSessionCompat != null) {
+            mMediaSessionCompat.setActive(false);
+            mMediaSessionCompat.release();
+        }
+        mMediaSessionCompat = new MediaSessionCompat(context, TAG);
+        mMediaSessionCompat.setFlags(MediaSessionCompat.FLAG_HANDLES_MEDIA_BUTTONS);
+        mMediaSessionCompat.setCallback(new BikeyMediaButtonCallback(context));
+        mMediaSessionCompat.setActive(true);
+    }
+
+    @MainThread
+    public void stop() {
+        Log.d();
+        if (mMediaSessionCompat != null) {
+            mMediaSessionCompat.setActive(false);
+            mMediaSessionCompat.release();
+            mMediaSessionCompat = null;
+        }
+    }
+
+    private static class BikeyMediaButtonCallback extends MediaSessionCompat.Callback {
+
+        private final Context mContext;
+
+        BikeyMediaButtonCallback(Context context) {
+            mContext = context;
+        }
+
+        @Override
+        public boolean onMediaButtonEvent(Intent intent) {
+            Log.d("intent=" + StringUtil.toString(intent));
```

Comment:
Info: the contents of this method haven't changed from the old `MediaButtonReceiver.onReceive()` method.